tests/flow/test_list.py

```diff
@@ -143,3 +143,633 @@ def test03_null_list_function_inputs(self):
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+        query = """MATCH (n:X) RETURN toBooleanList([n])"""
+        expected_result = []
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+        query = """MATCH ()-[e]->() RETURN toBooleanList(e)"""
+        expected_result = []
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+        query = """MATCH ()-[e]->() RETURN toBooleanList([e])"""
+        expected_result = []
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+        query = """MATCH p=()-[]->() RETURN toBooleanList(p)"""
+        expected_result = []
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+        query = """MATCH p=()-[]->() RETURN toBooleanList([p])"""
+        expected_result = []
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+        # Test of type mismatch input
+        try:
+            redis_graph.query("RETURN toBooleanList(true)")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Boolean", str(e))
+
+        try:
+            redis_graph.query("RETURN toBooleanList(7.05)")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Float", str(e))
+
+        try:
+            redis_graph.query("RETURN toBooleanList(7)")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Integer", str(e))
+
+        try:
+            redis_graph.query("RETURN toBooleanList('abc')")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was String", str(e))
+
+        try:
+            query = """CREATE (a:X)"""
+            redis_graph.query(query)
+            query = """MATCH (n) RETURN toBooleanList(n)"""
+            actual_result = redis_graph.query(query)
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List but was Node", str(e))
+            query = """MATCH (a:X) DELETE a"""
+            redis_graph.query(query)
+
+        try:
+            query = """CREATE (a:X), (b:Y)"""
+            redis_graph.query(query)
+            query = """MATCH (a:X), (b:Y) create (a)-[r:R]->(b)"""
+            redis_graph.query(query)
+            query = """MATCH ()-[e]->() RETURN toBooleanList(e)"""
```

Comment:
consider compacting this into a single query:
```
query = "CREATE (a:X)-[r:R]->(b:Y) RETURN toBooleanList(r)"
```

Reply:
Thanks. Done in [b36f30d](https://github.com/RedisGraph/RedisGraph/pull/2605/commits/b36f30dfbf0af4709ba9aa7c5ba4b01be17b10fd)